api/v1/slurmcluster_types.go

```diff
@@ -353,16 +353,17 @@ type PasswordSecretKeyRef struct {
 	Key string `json:"key"`
 }
 
-type MariaDbOpeator struct {
+type MariaDbOperator struct {
 	// +kubebuilder:validation:Optional
```

Comment:
This validation marker does nothing in terms of validation. It's just a decorative stuff, and redundant in case of controller-gen usage.
https://book.kubebuilder.io/reference/markers#difference-between--optional-and--kubebuildervalidationoptional

If you want a field to be optional, you have to specify `omitempty` in json tags